application/org.openjdk.jmc.browser.attach/pom.xml

```diff
@@ -55,6 +55,7 @@
 					<compilerArgs>
 						<args>--add-exports</args>
 						<args>jdk.attach/sun.tools.attach=ALL-UNNAMED</args>
+						<args>-version</args><!-- see https://bugs.eclipse.org/bugs/show_bug.cgi?id=544016 -->
```

Comment:
This is really weird. I'm struggling with the add-eports setting in my build attempts as well. I'm using JDK11. I also find that there's a difference in behavior between maven versions. E.g. maven 3.8.1 behaves slightly better than 3.8.4. But I have to do this kind of changes in several pom files until I get the build to go through.
What's your maven versions?

Reply:
My version is 3.8.4 which is the current version of maven. But it works for me with JDK 11. So the culprit (in my opinion) is the p2-maven-plugin, which is the only other thing I changed besides the JDK version, or of course, the JDK itself, but this seems to be unlikely.

Reply:
I have to test the new p2-maven-plugin version with JDK 11.

Reply:
Could it be a tycho issue, again. Tycho overrides many lifecycle of Maven:

https://github.com/eclipse/tycho/blob/5c83ecd968eae37aa328020d228c8f03b2a3168c/tycho-maven-plugin/src/main/resources/META-INF/plexus/components.xml#L27-L29

```xml
<compile>
  org.eclipse.tycho:tycho-compiler-plugin:${project.version}:compile
</compile>
```

It can't be `p2-maven-plugin` anyway as it is only used to generate a p2 repository (that is used in the `.target` files), and it is only used in `releng/third-party`.

Reply:
I played around with it further. tycho uses JDT under the covers. And it seems to be an issue between Tycho JDT and JDK17. mvn -X shows nice parameters were passed to JDT but I didn't look into it. I will try to see what happens when switching the tycho compiler to javac. Don't know whether that's possible/a good idea...

Reply:
I don't think that's possible, tycho overrides many aspects of maven, typicaly for the compiler the tycho _overlay_ will get depdencies from other tycho plugin, not from maven APIs.

Reply:
Yup, you're right. I figured that out. I guess we have to live with the workaround. We can try once again with tycho 2.6.0 which should be out really soon. But if that doesn't change a thing we can only open an issue with tycho.

Reply:
The problem persists with tycho 2.6.0.

Reply:
By the way the issue seems gone with tycho 2.7.1.